compiler/src/main/java/org/qbicc/graph/SimpleBasicBlockBuilder.java

```diff
@@ -121,6 +121,12 @@ public void setExceptionHandlerPolicy(final ExceptionHandlerPolicy policy) {
         this.policy = policy;
     }
 
+    public void startMethod(List<ParameterValue> arguments) {
+        if (currentBlock != null) {
+            throw new IllegalStateException("begin() called before startMethod()");
+        }
```

Comment:
I'd rather put the check on `begin` because in most cases where this bug will appear, `startMethod` will never be called at all.